src/handlers/wire.rs

```diff
@@ -70,46 +48,39 @@ pub mod aggregation {
 
     impl Write for Signature {
         fn write(&self, buf: &mut impl BufMut) {
-            (self.signature.len() as u32).write(buf);
-            buf.put_slice(&self.signature);
+            self.signature.write(buf);
         }
     }
 
     impl Read for Signature {
         type Cfg = ();
 
         fn read_cfg(buf: &mut impl Buf, _: &()) -> Result<Self, Error> {
-            let len = u32::read(buf)? as usize;
-            if buf.remaining() < len {
-                return Err(Error::EndOfBuffer);
-            }
-            let mut signature = vec![0u8; len];
-            buf.copy_to_slice(&mut signature);
+            let signature = Vec::<u8>::read_range(buf, ..)?;
```

Comment:
If you don't know how long the signature is, you might want to anyway bound it at some "reasonable" value to prevent a dos attack; if the sender says "hey I'm about to send you a vector that's 10Billion bytes long" then you will allocate that memory

Reply:
very good point. gonna introduce a `const MAX_SIGNATURE_SIZE_BYTES` and set it to 256 for now.